src/modules/uxrce_dds_client/uxrce_dds_client.cpp

```diff
@@ -345,18 +346,28 @@ bool UxrceddsClient::setup_session(uxrSession* session)
 
 	uxr_set_request_callback(session, on_request, this);
 
-	// Spin until sync with the Agent
+	uint8_t sync_timeouts = 0;
+
+	// Spin until sync with the Agent or the session sync has multiple timeouts
 	while (_synchronize_timestamps) {
-		if (uxr_sync_session(session, 1000) && _timesync.sync_converged()) {
-			PX4_INFO("synchronized with time offset %-5" PRId64 "us", session->time_offset / 1000);
+		if (uxr_sync_session(session, 1000)) {
+			if (_timesync.sync_converged()) {
+				PX4_INFO("synchronized with time offset %-5" PRId64 "us", session->time_offset / 1000);
 
-			if (_param_uxrce_dds_syncc.get() > 0) {
-				syncSystemClock(session);
-			}
+				if (_param_uxrce_dds_syncc.get() > 0) {
+					syncSystemClock(session);
+				}
 
-			break;
+				break;
+			}
+			sync_timeouts = 0;
+		} else {
+			sync_timeouts++;
 		}
 
+		if (sync_timeouts > TIMESYNC_MAX_TIMEOUTS) {
```

Comment:
Can we move this earlier before the pubs are initialized and the repliers (eg VehicleCommand) are created (dynamic allocation)? 

There's also `_connected = true` being set on line 337.

Reply:
Yes, this is possible and makes sense. I changed the order.